packages/web/components/bridge/immersive-bridge.tsx

```diff
@@ -36,15 +37,41 @@ export const ImmersiveBridgeFlow = ({
 }: PropsWithChildren<BridgeFlowProvider>) => {
   const { t } = useTranslation();
   const { isMobile } = useWindowSize();
+  const { logEvent } = useAmplitudeAnalytics();
 
   const [isVisible, setIsVisible] = useState(false);
   const [step, setStep] = useState<ImmersiveBridgeScreen>(
     ImmersiveBridgeScreen.Asset
   );
-  const [direction, setDirection] = useState<"deposit" | "withdraw">("deposit");
-  const { logEvent } = useAmplitudeAnalytics();
+  const [direction, setDirection] = useQueryState<
+    "deposit" | "withdraw" | null
+  >("transferDirection", {
+    history: "replace",
+    parse: (value) => {
+      if (value === "deposit") return "deposit";
+      if (value === "withdraw") return "withdraw";
+      return null;
+    },
+  });
+  const [selectedAssetDenom, setSelectedAssetDenom] = useQueryState<
+    string | null
+  >("transferAsset", {
+    defaultValue: null,
+    history: "replace",
+    parse: (value) => {
+      if (typeof value === "string") return value;
+      return null;
+    },
+  });
 
-  const [selectedAssetDenom, setSelectedAssetDenom] = useState<string>();
+  useEffect(() => {
+    if (!isNil(direction) && !isVisible) setIsVisible(true);
+    if (isNil(direction) && isVisible) setDirection("deposit"); // default direction
```

Comment:
I think we can set this in the `useQueryState` directly. defaultValue: "deposit" and parse returns "deposit" with invalid value

Reply:
I tried this, but then the state being set causes the flow to become visible. How do we differentiate that the value was the default or was set from query params?